683 update access esm how to run

blimlim opened the pull request:
## Description

This pull request updates the [run access esm tutorial](https://access-hive.org.au/models/run-a-model/run-access-esm/) to match the structure of the OM2 tutorial, and to include changes related to the latest payu updates and the (to be) released configurations.

The new _ESM how to run_ page was set up by copying the OM2 page, and modifying the information to be correct for ESM1.5.

This is not currently ready to merge, as some changes cannot be made until the final release, specifically:
- The link to the release notes will have to wait until there is a release notes post on the forum.
- The required payu version might need to wait until the final updates to payu are released.
- Links to the released configurations on github will have to wait until the `release-` branches are ready.
- The `postscript` example will have to wait for the NetCDF conversion to be finalised. 

In the mean time though, it would be great to get feedback and suggestions on the other changes!

Fixes #683 

## Type of change
- [x] New link / content

## Checklist:

- [x] The new content is accessible and located in the appropriate section
- [x] My changes do not break navigation and do not generate new warnings
- [x] I have checked that the links are valid and point to the intended content
- [ ] I have checked my code/text and corrected any misspellings
- [ ] I have chosen the correct tag for the level of [support provided](https://access-hive.org.au/#support)

blimlim commented:
Hi everyone, I've finally finished adding in some of the more ESM1.5 specific information now that the changes to the configurations and payu are (mostly) finalised and I think it's ready for a review. 

The following two points are still waiting on further information/links, but everything else should be ready for a review:

- The required payu version might need to wait until the final updates to payu are released.
- Links to the released configurations on github will have to wait until the release- branches are ready.

Sorry about the delay on this!

KAUR1984 commented:
Thanks for that @blimlim! Looks great to me!

I would perhaps suggest to merge this PR into the development branch for now. For the rest of the changes, happy to create a new PR and once we have all the changes, we can deploy it on the main branch. :)

blimlim commented:
Thanks @KAUR1984! I just realised there were a couple of spelling errors which I've just fixed up, and then will merge into development

aidanheerdegen commented:
I'm part way through reviewing. Should I just close off that review and allow you to make the suggested changes?

blimlim commented:
That sounds good! I'll wait until I've added any suggested changes before merging.

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
 
 - **Run 7 years of simulation with resubmission every 3 years**<br>
-    To have a _total experiment length_ of 7 years with a 3-year resubmission cycle, leave [`runtime`](#runtime) as the default value of `1 year` and set [`runspersub`](#runspersub) to `3`. Then, run the configuration with `-n` set to `7`:
+    To have a _total experiment length_ of 7 years with a 3-year resubmission cycle, leave [`runtime`](#runtime) as the default value of `1 year` and set `runspersub` to `3`. Then, run the configuration with `-n` set to `7`:
```
```suggestion
    To have a _total experiment length_ of 7 years with a 3-year resubmission cycle, leave [`runtime`](#runtime) as the default value of `1 year`, set `runspersub` to `3` and `walltime` to `6:00:00`. Then, run the configuration with `-n` set to `7`:
```

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
-    
-- **Run 1 year and 4 months of simulation with resubmission every 4 months**<br>
-    To have a _total experiment length_ of 1 year and 4 months (16 months), you need to split it into multiple runs. For example, 4 runs of 4 months each. In this case, set the [`runtime`](#runtime) as follows:
+
```
```suggestion
!!! tip
    The `walltime` must be set to be long enough that the PBS job can complete. The model usually runs a single year in 90 minutes or less, but the `walltime` for a single model run is set to `2:30:00` out of an abundance of caution to make sure the model has time to run when there are occasional slower runs for unpredicatable reasons. When setting `runspersub > 1` the `walltime` doesn't need to be a simple multiple of `2:30:00` because it is highly unlikely that there will be multiple anomalously slow runs per submit.
```

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
````diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
+        days: 0
+```
+
+With the default configuration settings, the sea ice component of {{ model }} will produce restart files only at the end of each year. If valid restart files are required when running shorter simulations, the sea ice model configuration should be modified so that restart files are produced at monthly frequencies. To do this, change the `dumpfreq = 'y'` setting to `dumpfreq = 'm'` in the `cice_in. ...
````
```suggestion
With the default configuration settings, the sea ice component of {{ model }} will produce restart files only at the end of each year. If valid restart files are required when running shorter simulations, the sea ice model configuration should be modified so that restart files are produced at monthly frequencies. To do this, change the `dumpfreq = 'y'` setting to `dumpfreq = 'm'` in the `cice_in.nml` configuration file located in the `ice` subdirectory of the control directory.
```

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
+
+These lines can be edited to change the [PBS directives](https://opus.nci.org.au/display/Help/PBS+Directives+Explained) for the [PBS job][PBS job].
+
+For example, to run {{ model }} under the `ol01` project (COSIMA Working Group), uncomment the line beginning with `# project` by deleting the `#` symbol and replace `PROJECT_CODE` with `ol01`:
```
```suggestion
By default the model will be submitted to the PBS queue using your default project.  To run {{ model }} using the resources of a specific project, for example the `lg87` project (ESM Working Group), uncomment the line beginning with `# project` by deleting the `#` symbol and replace `PROJECT_CODE` with `lg87`:
```

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
````diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
+For example, to run {{ model }} under the `ol01` project (COSIMA Working Group), uncomment the line beginning with `# project` by deleting the `#` symbol and replace `PROJECT_CODE` with `ol01`:
+
+```yaml
+project: ol01
````
```suggestion
project: lg87
```

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
````diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
+```
+
+!!! warning
+    If a project other than the user's default project is used to run {{ model }} configuration, then the `shortpath` field also needs to be uncommented and the path to the desired `/scratch/PROJECT_CODE` added.<br>
+    Doing this will make sure the same `/scratch` location is used for the _laboratory_, regardless of which project is used to run the experiment.
````
```suggestion
    If more than one project is used to run an {{ model }} configuration the `shortpath` option also needs to be uncommented and the path to the desired `/scratch/PROJECT_CODE` directory added.<br>
    This ensures the same `/scratch` location is used for the _laboratory_, regardless of which project is used to run the experiment.
```

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
+To enable syncing, change `enable` to `True`, and set `path` to a location on `/g/data`, where _payu_ will copy output and restart folders. A sensible `path` could be: `/g/data/$PROJECT/$USER/{{model}}/experiment_name/`.
+
+!!! Warning
+    The {{model}} default configurations include a postprocessing [postcript](#postscripts) which converts atmospheric outputs to NetCDF format and runs in a separate PBS job. This prevents the final run's output and restart files from being automatically synced. Once the postprocessing is completed, the final outputs and restarts can be manually synced by running `payu sync` from the control dir ...
```
```suggestion
    The {{model}} configurations include a postprocessing [postcript](#postscripts) which converts atmospheric outputs to NetCDF format. This runs in a separate PBS job and this prevents the output and restart files of the most recent run from being automatically synced. When a series of runs completes and the final post-processing is completed run `payu sync` in the control directory to sync the final outputs and restarts.
```

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
-## Monitor {{ model }} runs
+#### Collate
+
+Rather than outputting a single diagnostic file over the whole model horizontal grid, the ocean component [MOM](/models/model_components/ocean/#modular-ocean-model-mom) typically generates diagnostic outputs as tiles, each of which spans over a portion of the model horizontal grid.
```
```suggestion
Rather than outputting a single diagnostic file over the whole model horizontal grid, the ocean component [MOM](/models/model_components/ocean/#modular-ocean-model-mom) typically generates diagnostic outputs as tiles, each of which spans a portion of model grid.
```

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
````diff
@@ -363,108 +338,323 @@ With the correct use of [`runtime`](#runtime), [`runspersub`](#runspersub) and `
 
-If there are no jobs listed with your `jobname` (or if no job is listed), your run either successfully completed or was terminated due to an error.<br>
-For more information, check [NCI documentation](https://opus.nci.org.au/display/Help/FAQ+1%3A+Why+My+Jobs+are+NOT+Running).
+A dictionary to run scripts or subcommands at various stages of a _payu_ submission. The above example comes from the `release-historical+concentrations` configuration, where the ```update_landuse_driver.sh``` is used to apply historical land use changes at the end of each run.
````
```suggestion
Run scripts or subcommands at various stages of a _payu_ submission. The above example comes from the `release-historical+concentrations` configuration, where the ```update_landuse_driver.sh``` is used to apply historical land use changes at the end of each run.
```

aidanheerdegen reviewed (changes requested):
Thanks @blimlim! Awesome work.

I have made a number of suggestions, not all are necessarily required, happy to leave it to you to decide what to keep and what to not, or further modify.

I have some thoughts about the post-processing, but that may require changes to the script invocation, which will then have to be changed here to be consistent.

blimlim commented:
Thanks for spotting those things – I think they're all good suggestions and I've added them in. 

> I have some thoughts about the post-processing, but that may require changes to the script invocation, which will then have to be changed here to be consistent.

Sounds good, lets see if we can simplify how the script is called.

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
````diff
@@ -1,470 +1,662 @@
+    enable: False # set path below and change to true
+    path: null # Set to location on /g/data or a remote server and path (rsync syntax)
+```
+To enable syncing, change `enable` to `True`, and set `path` to a location on `/g/data`, where _payu_ will copy output and restart folders. A sensible `path` could be: `/g/data/$PROJECT/$USER/{{model}}/experiment_name/`.
````
```suggestion
To enable syncing, change `enable` to `True`, and set `path` to a location on `/g/data`, where _payu_ will copy output and restart folders.
```

We don't want users literally putting environment variables in the sync path, so I've suggested we just remove that bit as it got a little confusing trying to suggest something with a bunch of placeholders that needed to be explained and replaced for limited utility.

blimlim replied:
Good idea, I've just added this in. This section was copied from the OM2 docs – should I add an issue to add the same update there?

blimlim commented:
I've just updated the docs with the change to the NetCDF conversion behaviour and added in the correct payu version. After looking over everything, I think the information is now correct.

aidanheerdegen reviewed (approved):
Looks great, thanks for all the hard work.

atteggiani commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -1,470 +1,662 @@
-{% set model = "ACCESS-ESM" %}
+{% set model = "ACCESS-ESM1.5" %}
```
Solved in my recent [commit](https://github.com/ACCESS-NRI/access-hive.org.au/pull/726/commits/76bdfbfa2e885d412914e9b84e218b960ad752af)

atteggiani commented on docs/models/run-a-model/run-access-esm.md:
````diff
@@ -1,470 +1,662 @@
+
+The length of an {{model}} run is controlled by the `runtime` settings in the `config.yaml` file:
 
 ```
````
Solved in my recent [commit](https://github.com/ACCESS-NRI/access-hive.org.au/pull/726/commits/76bdfbfa2e885d412914e9b84e218b960ad752af)

atteggiani commented on docs/models/run-a-model/run-access-esm.md:
````diff
@@ -1,470 +1,662 @@
+### Running for less than one year {: id="shorter-runs"}
+When debugging changes to a model, it is common to reduce the run length to minimise resource consumption and return faster feedback on changes. In order to run the model for a single month, the `runtime` can be changed to
+
+```
````
Solved in my recent [commit](https://github.com/ACCESS-NRI/access-hive.org.au/pull/726/commits/76bdfbfa2e885d412914e9b84e218b960ad752af)

atteggiani commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -1,470 +1,662 @@
-If you want to manually terminate a run, you can do so by executing:
+
+#### Postscripts {: id="postscripts"}
+Postprocessing scripts that need to be run after _payu_ has completed all steps that might alter the output directory can be run as postscripts. These run in separate PBS jobs than the main model simulation. 
```
Solved in my recent [commit](https://github.com/ACCESS-NRI/access-hive.org.au/pull/726/commits/76bdfbfa2e885d412914e9b84e218b960ad752af)

atteggiani commented on docs/models/run-a-model/run-access-esm.md:
````diff
@@ -1,470 +1,662 @@
+postscript: -v PAYU_CURRENT_OUTPUT_DIR,PROJECT -lstorage=${PBS_NCI_STORAGE} ./scripts/NetCDF-conversion/UM_conversion_job.sh
 ```
-qdel <job-ID>
+All {{ model }} configurations include the above NetCDF conversion postscript, which converts the atmosphere model's fields file format output to NetCDF in order to aid analysis and reduce storage requirements. By default, the conversion script will delete the fields files upon successful completion, leaving only the NetCDF output.  This automatic deletion can be disabled by commenting out the `- ...
````
Solved in my recent [commit](https://github.com/ACCESS-NRI/access-hive.org.au/pull/726/commits/76bdfbfa2e885d412914e9b84e218b960ad752af)

aidanheerdegen commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -93,7 +93,7 @@ to a new experiment branch (`-b expt`) to a directory named `preindustrial+conce
     <terminal-line>Updated metadata. Experiment UUID: 0635396b-678d-45f9-b81e-abd25a2d7bf0</terminal-line>
     <terminal-line>Added archive symlink to /scratch/.../access-esm/archive/preindustrial+concentrations-expt-0635396b</terminal-line>
-    <terminal-line>To change directory to control directory run:</terminal-line>
+    <terminal-line>To change directory to _control_ directory run:</terminal-line>
```
This formatting seems to break in the terminal emulator. The word has disappeared in the preview.

I assume this change was unintentional.

atteggiani replied:
Oops I just did a "replace all" and missed that that was inside a terminal window 😅 
Fixed!

aidanheerdegen replied:
I figured.

aidanheerdegen reviewed (changes requested):
Just one small change, otherwise looks good and thanks for the changes.

aidanheerdegen reviewed (approved):
Looks great. Thanks!
